docs/usage/enabling_immutable_snapshots.md

```diff
@@ -0,0 +1,372 @@
+           --account-name [STORAGE_ACCOUNT] \
+           --container-name [CONTAINER_NAME] \
+           --query etag --output tsv)
+   ```
+
+2. **Extend the Immutability Period**
+
+   ```bash
+   az storage container immutability-policy extend \
+       --resource-group [RESOURCE_GROUP] \
+       --account-name [STORAGE_ACCOUNT] \
+       --container-name [CONTAINER_NAME] \
+       --period [NEW_IMMUTABILITY_PERIOD] \
+       --if-match $etag
+   ```
+
+   **Example:**
+
+   ```bash
+   az storage container immutability-policy extend \
+       --resource-group myResourceGroup \
+       --account-name myStorageAccount \
+       --container-name my-container \
+       --period 7 \
+       --if-match $etag
+   ```
+
+3. **Delete an Unlocked Policy**
+
+   ```bash
+   az storage container immutability-policy delete \
+       --resource-group [RESOURCE_GROUP] \
+       --account-name [STORAGE_ACCOUNT] \
+       --container-name [CONTAINER_NAME] \
+       --if-match $etag
+   ```
+
+### Lock the Immutability Policy
+
+Locking the immutability policy makes it irreversible and ensures that the policy cannot be reduced or removed. This provides compliance with regulatory requirements.
+
+#### Google Cloud Storage (GCS)
+
+To lock the immutability policy:
+
+> [!CAUTION]
+> Locking an immutability policy is an irreversible action.
+
+1. **Lock the Policy**
+
+   ```bash
+   gcloud storage buckets update gs://[BUCKET_NAME] \
+       --lock-retention-period
+   ```
+
+   **Example:**
+
+   ```bash
+   gcloud storage buckets update gs://my-bucket \
+       --lock-retention-period
+   ```
+
+
+#### Azure Blob Storage (ABS)
+
+To lock the immutability policy:
+
+> [!WARNING]
+> Once you have thoroughly tested a time-based immutability policy, you can proceed to lock the policy. A locked policy ensures compliance with regulations such as SEC 17a-4(f) and other regulatory requirements. While you can extend the immutability interval for a locked policy up to **_`five times`_**, it cannot be shortened.
+> After a policy is locked, it cannot be deleted. However, you can delete the blob once the immutability interval has expired.
+
+1. **Retrieve the Policy's ETag**
+
+   ```bash
+   etag=$(az storage container immutability-policy show \
+           --account-name [STORAGE_ACCOUNT] \
+           --container-name [CONTAINER_NAME] \
+           --query etag --output tsv)
+   ```
+
+2. **Lock the Policy**
+
+   ```bash
+   az storage container immutability-policy lock \
+       --resource-group [RESOURCE_GROUP] \
+       --account-name [STORAGE_ACCOUNT] \
+       --container-name [CONTAINER_NAME] \
+       --if-match $etag
+   ```
+
+   **Example:**
+
+   ```bash
+   az storage container immutability-policy lock \
+       --resource-group myResourceGroup \
+       --account-name myStorageAccount \
+       --container-name my-container \
+       --if-match $etag
+   ```
+
+   **Caution:** Once locked, the policy cannot be removed or decreased.
```

Comment:
Can this be moved before the steps and changed to a `[!CAUTION]` like it is for GCS too?